### DISCUSSION TITLE
Artisan migrate:refresh should not run without warning even in app_env=DEV

### DISCUSSION THREAD

**Discussion by ataldev** [laravel/framework#53801] [General] [answered]
This happened to me multiple times and it's a mess to fix everytime. We have multiple databases and the over 200 migrations in our app. I am using Amazon Q assistant in the terminal, which gives suggestions to finish your commands before you finish them. 
Multiple times I was trying to run `migrate:rollback` , but after typing `migrate:r` hitting tab and enter, I've lost all my local changes and the database is fresh. 
Let me know of your suggestions on how I can prevent this?

**Comment by rojtjo** [comment, marked as answer]
You could mark the commands as prohibited, but that would prevent you from running them all together.

```php
use Illuminate\Database\Console\Migrations\RefreshCommand;
use Illuminate\Support\Facades\DB;

// Prevent destructive DB commands
Illuminate\Support\Facades\DB::prohibitDestructiveCommands(true);

// Or just the `migrate:refresh` command
RefreshCommand::prohibit(true);
```

See #51376

**Reply by ataldev** [reply]
Thanks.

**Comment by ghost** [comment]
Finally someone complaining about a copilot. This is why I avoid using it.
